### PR TITLE
core: frontend: Add notifier

### PR DESCRIPTION
**Pull request by patrickelectric**
- Encapsulates notification logic
- Remove duplicated code
- Fixes backend offline checks
- Simplify code for notification and services

### CODE REVIEW

File: core/frontend/src/libs/notifier.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import notifications from '@/store/notifications'
```
Comment (rafaellehmkuhl) [inline core/frontend/src/libs/notifier.ts:1]: I liked the simplification in use. Fewer lines to do the same thing.
Didn't get why this is separated from the store, though. Seems to me, they have the same scope and this is just a helper, so would be better in the same file.

**Review by rafaellehmkuhl — COMMENTED** [review]
Could you squash the 31 commits that do the same thing?

File: core/frontend/src/components/app/PowerMenu.vue [hunk 2]
```diff
@@ -185,8 +187,7 @@ export default Vue.extend({
           if (error.code === 'ECONNABORTED') {
             return
           }
-          const message = error.response?.data?.detail ?? error.message
-          notifications.pushError({ service: commander_service, type: 'SHUTDOWN_FAIL', message })
+          notifier.pushBackError('SHUTDOWN_FAIL', error)
```
Comment (joaoantoniocardoso) [inline core/frontend/src/components/app/PowerMenu.vue:190]: Couldn't 'BACKEND_OFFLINE' and etc be enumerations?

File: core/frontend/src/libs/notifier.ts (hunk 1 above)
Comment (patrickelectric) [reply core/frontend/src/libs/notifier.ts:1]: there is a rule on the project that does not allow multiple classes definitions on the same file.
https://eslint.org/docs/rules/max-classes-per-file

**Comment by patrickelectric** [comment]
> Could you squash the 31 commits that do the same thing?

That may result in huge conflicts for people that have open PRs or WIP branches.

**Comment by patrickelectric** [comment]
> > Could you squash the 31 commits that do the same thing?
> 
> That may result in huge conflicts for people that have open PRs or WIP branches, that can also affect the rebase of this PR if something merged conflicts with it.

Note: This PR is still in draft.

File: core/frontend/src/components/app/PowerMenu.vue (hunk 2 above)
Comment (patrickelectric) [reply core/frontend/src/components/app/PowerMenu.vue:190]: That's indeed possible, but this is an error string for logging and human/machine readable.
Maintaining the enum may be problematic, share it between all services, having each service an enum, share and deal with multiple types for typescript and etc. There is no advantage doing such thing.

**Comment by rafaellehmkuhl** [comment]
> > Could you squash the 31 commits that do the same thing?
> 
> That may result in huge conflicts for people that have open PRs or WIP branches, that can also affect the rebase of this PR if something merged conflicts with it. 

It will be literally the same conflict to be solved. 

Its a little bizarre you're separating this into 31 commits different commits since it's the exactly the same logic change in all of them. You've taught me that in fact.

We squash logic changes since day one of this repository. What changed?

**Comment by patrickelectric** [comment]
> > > Could you squash the 31 commits that do the same thing?
> > 
> > 
> > That may result in huge conflicts for people that have open PRs or WIP branches, that can also affect the rebase of this PR if something merged conflicts with it.
> 
> It will be literally the same conflict to be solved.
> 
> Its a little bizarre you're separating this into 31 commits different commits since it's the exactly the same logic change in all of them. You've taught me that in fact.
> 
> We squash logic changes since day one of this repository. What changed?

We always have been broken commits per frontend/folder/struct, this just breaks in files to make it easy to rebase. But if you are ok to fixing such conflicts.

**Comment by rafaellehmkuhl** [comment]
> > > > Could you squash the 31 commits that do the same thing?
> > > 
> > > 
> > > That may result in huge conflicts for people that have open PRs or WIP branches, that can also affect the rebase of this PR if something merged conflicts with it.
> > 
> > 
> > It will be literally the same conflict to be solved.
> > Its a little bizarre you're separating this into 31 commits different commits since it's the exactly the same logic change in all of them. You've taught me that in fact.
> > We squash logic changes since day one of this repository. What changed?
> 
> We always have been broken commits per frontend/folder/struct, this just breaks in files to make it easy to rebase. But if you are ok to fixing such conflicts.

I'm very ok. We had [the same case](https://github.com/bluerobotics/BlueOS-docker/pull/969) yesterday, in fact. Easy PR. Two commits. No big deal.

**Review by rafaellehmkuhl — APPROVED** [review]
Great change. Simplify notifications usage a lot.

**Pull request merged**